lib/CL/devices/devices.c

```diff
@@ -431,9 +431,11 @@ pocl_init_devices()
             pocl_devices[dev_index].available = 0;
           }
 
-          if (dev_index == 0)
-            pocl_devices[dev_index].type |= CL_DEVICE_TYPE_DEFAULT;
-
+          if (pocl_get_bool_option ("POCL_WORKAROUND_DEVTYPE", 0) == 0)
```

Comment:
This is actually a workaround, we **do** set the CL_DEVICE_TYPE_DEFAULT by default :) this bool option disables ORing the CL_DEVICE_TYPE_DEFAULT, so software that treats device_type as enum (checks with equality) instead of bitfield, will work.

I've added some docs and renamed the option to POCL_WORKAROUND_DEVICE_TYPE_BUG.